src/v2/view-builder/views/idp/AuthenticatorIdPVerifyView.js

```diff
@@ -42,21 +38,11 @@ const Body = BaseForm.extend({
   },
 
   save() {
-    return this.options.isChallenge
-      ? loc('mfa.challenge.verify', 'login')
-      : loc('mfa.enroll', 'login');
+    return loc('mfa.challenge.verify', 'login');
   },
 
 });
 
 export default BaseAuthenticatorView.extend({
-  initialize() {
-    BaseAuthenticatorView.prototype.initialize.apply(this, arguments);
-
-    // The IdP Authenticator doesn't have a traditional enroll/challenge form.
-    // We need to infer it based on if currentAuthenticatorEnrollment is included in the response
-    const currentAuthenticatorEnrollment = this.options.appState.get('currentAuthenticatorEnrollment');
-    this.options.isChallenge = !!currentAuthenticatorEnrollment;
-  },
```

Comment:
My motivation for keeping this as a single file was to reduce the duplicate code between enroll/challenge views. Some authenticators (Duo, IdP, Symantec) have nearly identical code, only changing the title, subtitle, and save form labels.

This may have been an over-optimization to condense code, but let's have a conversation about ways to solve this problem. At a high level:

- Authenticators always have an enroll/challenge view state. Ideally, the title, subtitle, and save form labels (`Enroll in {0}`/`Verify with {0}`, `Enroll`/`Verify`) should be constant throughout all authenticators.
- Form and top-level (`messages` present in response) error messages should be shown when present. This is opt-in today, as errors render as basic text on the page.
- Header/footer logic is also relatively constant throughout all authenticators.
- The decision to show the enroll or challenge page can be inferred based on whether there is a `currentAuthenticatorEnrollment`. This has been moved to an `appState` helper called [`isAuthenticatorChallenge `](https://github.com/okta/okta-signin-widget/blob/ff47bcfd36a5a44d3fff465c6407ba883acb774e/src/v2/models/AppState.js#L122-L129).

Reply:
It's sort of deficit the dispatch logic in ViewFactory (hence going to be a little redundant work)
The ViewFactory already has such logic like 
- `if is challenge, use ViewA; if is enroll, use ViewB`

What you suggest here is to push down the dispatch one level down. (see the `if is challenge else` runs twice)
- In ViewFactory, `if is challenge, use ViewA; if is enroll, use ViewA`
- In ViewA, `if is challenge, does X; if is enroll, does B`

In this draft, The `messages` method is duplicated but it's easy to get rid of via inheritance: creates a base class for IdPAuthenticator.

Reply:
Doesn't seem conflict with your goal to change more general label like `Verify with {0}`? 
I'm thinking we would have

- BaseEnrollAuthenticatorForm
- BaseVerifyAuthenticatorForm

which take care of Title/Subtitle/Save for all authenticators.

For handling `messages`, hopefully we could take care it BaseForm/BaseView with maybe couple of special case in individual View level.

Reply:
☝🏼 These are all excellent points and resolves a variety of problems. Let's scope the inheritance modeling work for another time!